haskell/repl.bzl

```diff
@@ -316,12 +316,31 @@ def _compiler_flags_and_inputs(hs, repl_info, static = False, path_prefix = ""):
         cc_library_files = _concat(get_library_files(hs, cc_libraries_info, cc_libraries))
     else:
         cc_library_files = get_ghci_library_files(hs, cc_libraries_info, cc_libraries)
-    link_libraries(cc_library_files, args)
+
```

Comment:
@aherrmann thoughts on this? I considered patching `link_libraries` but couldn't see a nice way to do it that didn't make it kind of pointlessly general (e.g. I started by trying `lib_format = "-l%s"` / `dir_format = "-L%s"` flags to supplant `prefix_optl` but that doesn't really work because you need to conditionally format and if I pass in lambdas, what is the function really abstracting for me?). Thus I implemented a specialised version here. Moreover, the use of `bazel-out` feels bit ugly but I couldn't see an obviously nicer way -- perhaps if this issue only affects external repositories we could instead test for `.startswith("external")` and flip the condition? I don't have enough confidence to say either way; thoughts welcome.

Reply:
A good way to do this may be an additional `path_prefix = ""` optional argument to `link_libraries`. The `-L` format strings could then be extended with that, e.g.
```
    dirfmt = "-optl-L" + path_prefix + "%s"
    ...
    dirfmt = "-L" + path_prefix + "%s"
```
This doesn't have to be conditional on whether the path starts with `bazel-out` or `external`. Both these types of paths exist under the execroot.